clinical-domain-agent/src/test/java/care/smith/fts/cda/impl/TCACohortSelectorIT.java

```diff
@@ -0,0 +1,211 @@
+
+  private static final String PID_SYSTEM =
+      "https://ths-greifswald.de/fhir/gics/identifiers/Pseudonym";
+  private static final String POLICY_SYSTEM =
+      "https://ths-greifswald.de/fhir/CodeSystem/gics/Policy";
+
+  private static TCACohortSelector cohortSelector;
+
+  private static MockCohortSelector allCohortSelector;
+  private static MockCohortSelector listCohortSelector;
+
+  @BeforeEach
+  void setUp(WireMockRuntimeInfo wireMockRuntime, @Autowired WebClientFactory clientFactory) {
+    var address = "http://localhost";
+    var server = new HttpClientConfig(address);
+    var config = new TCACohortSelectorConfig(server, PID_SYSTEM, POLICY_SYSTEM, POLICIES, "MII");
+    cohortSelector =
+        new TCACohortSelector(
+            config, clientFactory.create(clientConfig(wireMockRuntime)), meterRegistry);
+    wireMock = wireMockRuntime.getWireMock();
+    allCohortSelector = MockCohortSelector.fetchAll(wireMock, POLICY_SYSTEM);
+    listCohortSelector = MockCohortSelector.fetch(wireMock, POLICY_SYSTEM);
+  }
+
+  private static MappingBuilder getBuilderFetchAll() {
+    return post("/api/v2/cd/consented-patients/fetch-all")
+        .withHeader(CONTENT_TYPE, equalTo(APPLICATION_JSON_VALUE));
+  }
+
+  private static MappingBuilder getBuilderFetch() {
+    return post("/api/v2/cd/consented-patients/fetch")
+        .withHeader(CONTENT_TYPE, equalTo(APPLICATION_JSON_VALUE));
+  }
+
+  @Override
+  protected Stream<TestStep<?>> createTestSteps() {
+    return Stream.of(
+        new TestStep<ConsentedPatient>() {
+          @Override
+          public MappingBuilder getBuilder() {
+            return TCACohortSelectorIT.getBuilderFetchAll();
+          }
+
+          @Override
+          public Flux<ConsentedPatient> executeStep() {
+            return TCACohortSelectorIT.cohortSelector.selectCohort(List.of());
+          }
+        },
+        new TestStep<ConsentedPatient>() {
+          @Override
+          public MappingBuilder getBuilder() {
+            return TCACohortSelectorIT.getBuilderFetch();
+          }
+
+          @Override
+          public Flux<ConsentedPatient> executeStep() {
+            return TCACohortSelectorIT.cohortSelector.selectCohort(List.of("id"));
+          }
+
+          @Override
+          public String acceptedContentType() {
+            return APPLICATION_JSON_VALUE;
+          }
+        });
+  }
+
+  @Test
+  void responseInvalidErrors() {
+    wireMock.register(getBuilderFetchAll().willReturn(ok().withBody("invalid")));
+    create(cohortSelector.selectCohort(List.of())).expectError().verify();
+  }
+
+  @Test
+  void badRequestErrors() throws JsonProcessingException {
+    var om = new ObjectMapper();
+    wireMock.register(
+        getBuilderFetchAll()
+            .willReturn(
+                badRequest()
+                    .withHeader("Content-Type", APPLICATION_JSON_VALUE)
+                    .withBody(
+                        om.writeValueAsString(
```

Comment:
We use fhirResourceToString in other places, maybe we should extract a similar method for generic json.